### DISCUSSION TITLE
How to import from local file?

### DISCUSSION THREAD

**Discussion by jakepm** [engram-design/FeedMe#960] [Q&A]
### Description

"Feed URL
**Can be a file on the local file system** or from a website url (http://...)."

Have tried putting the file in /web/ and in the root directory, but the plugin just keeps repeating how it cannot find the file. Could you please state exactly where this file is supposed to go, as using Curl is not suitable in many cases due to environment limitations.

### Steps to reproduce

1. Install plugin
2. Use plugin to try and import local file

### Additional info

- Plugin version:
- Craft version:

**Comment by wblommaert** [comment]
@jakepm When entering the feed location you can provide an absolute path to the file on your system. Because those paths can vary between environments it's best to set up an alias (if you haven't already) which can then be used when specifying the file location.

Say for example you have a feeds folder on your server under `/var/www/private_html/feeds/`, you could make an alias `@feeds` that points to said location. Next you can enter `@feeds/feed.xml` when setting up a new feed in FeedMe. Alternatively you could just specify the absolute path directly, but this makes it less flexible across environments. Using the alias you can set up a path per environment or fetch it from an environment variable depending on your setup.

**Comment by jakepm** [comment]
Ah, I didn't realise that the path had to be absolute and assumed it was from the project root or web root. I'll follow you're guidance and am sure I'll be fine from here. Thank you for your help! :)

**Comment by ghost** [comment]
An easier approach (maybe) that is to put the feeds into storage folder. 
So you can just use something like `@storage/feeds/importfile.xml`

[Referring to nystudio: ](https://nystudio107.com/blog/multi-environment-configuration-for-craft-cms-3)
> Craft CMS and Yii2 define a num­ber of alias­es for you already, that you’re free to use if you like:
`@storage — path to the storage/ direc­to­ry in your project`
